File: libs/core/src/main/java/com/example/iter/common/audit/service/AdminActionService.java
package com.example.iter.common.audit.service;

import com.example.iter.common.audit.domain.entity.AdminAction;
import com.example.iter.common.audit.domain.entity.AdminActionTargetType;
import com.example.iter.common.audit.domain.entity.AdminActionType;
import com.example.iter.common.audit.domain.repository.AdminActionRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

// 관리자 도메인(회원/장비/분쟁 관리 등)에서 조치를 취할 때마다 이 서비스를 호출해 감사 로그를 남긴다.
@Service
@RequiredArgsConstructor
public class AdminActionService {

    private final AdminActionRepository adminActionRepository;

    @Transactional
    public void record(Long adminId, AdminActionTargetType targetType, Long targetId, AdminActionType action, String reason) {
        AdminAction adminAction = AdminAction.builder()
                .adminId(adminId)
                .targetType(targetType)
                .targetId(targetId)
                .action(action)
                .reason(reason)
                .build();
        adminActionRepository.save(adminAction);
    }
}

File: libs/core/src/main/kotlin/com/example/iter/common/audit/service/AdminActionService.kt
package com.example.iter.common.audit.service

import com.example.iter.common.audit.domain.entity.AdminAction
import com.example.iter.common.audit.domain.entity.AdminActionTargetType
import com.example.iter.common.audit.domain.entity.AdminActionType
import com.example.iter.common.audit.domain.repository.AdminActionRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class AdminActionService(private val adminActionRepository: AdminActionRepository) {
    // 관리자 상태 변경 트랜잭션에 참여해 대상 변경과 감사 이력이 함께 커밋되거나 롤백되게 합니다.
    @Transactional
    fun record(adminId: Long, targetType: AdminActionTargetType, targetId: Long, action: AdminActionType, reason: String?) {
        adminActionRepository.save(
            AdminAction(
                adminId = adminId,
                targetType = targetType,
                targetId = targetId,
                action = action,
                reason = reason
            )
        )
    }
}
